ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+        self.flexy_mnt_container_dir = config.ENV_DATA.get(
+            'flexy_mnt_container_dir', constants.FLEXY_MNT_CONTAINER_DIR
+        )
+        # Path inside container where flexy_private_conf_dir will be mounted
+        # same value needs to be set for BUSHSLICER_PRIVATE_DIR in env file
+        # set BUSHSLICER_PRIVATE_DIR in ocs-osp.env itself
+        self.flexy_mnt_private_conf_dir = config.ENV_DATA.get(
+            'flexy_mnt_private_conf_dir', constants.FLEXY_MNT_PRIVATE_CONF_DIR
+        )
+        self.flexy_openshift_misc_url = config.ENV_DATA.get(
+            'flexy_openshift_misc_url', constants.OCP_QE_MISC_REPO
+        )
+        self.flexy_private_conf_url = config.ENV_DATA.get(
+            'flexy_private_conf_url', constants.FLEXY_PRIVATE_CONF_URL
+        )
+        self.flexy_img_url = config.ENV_DATA.get('flexy_img_url')
+        self.flex_env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
+        self.flexy_log_file = os.path.expanduser(
+            config.ENV_DATA.get('flexy_log_file')
+        )
+        self.exitThread = False
+
+    def run_container(self, cmd_string):
+        """
+        Actualy container run happens here, a thread will be
+        spawned to asynchronously print flexy container logs
+
+        """
+        logger.info(
+            f"Starting Flexy container with options {cmd_string}"
+        )
+        # Launch a seperate thread for getting the log file contents
+        # simultaneously while flexy container running else logging will be
+        # blocked till container terminates
+        with ThreadPoolExecutor(max_workers=1) as tp:
+            future = tp.submit(self.get_log)
+            cp = subprocess.run(
+                cmd_string, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+            if cp.returncode:
+                if cp.stderr:
+                    logger.error(cp.stderr)
+                raise exceptions.CommandFailed("Flexy run failed")
+            logger.info("Flexy run finished successfully")
+            self.exitThread = True
+        for res in TimeoutSampler(60, 1, future.done):
+            if not res:
+                logger.info("Waiting for logger thread to join")
+
+    def build_install_cmd(self):
+        """
+        Build flexy command line for 'deploy' operation
+
+        """
+        cmd = "sudo podman run --rm=true"
+        flexy_container_args = self.build_container_args()
+        cmd_string = " ".join([cmd, " ".join(flexy_container_args)])
+        return cmd_string
+
+    def build_destroy_cmd(self):
+        """
+        Build flexy command line for 'destroy' operation
+
+        """
+        cmd = "sudo podman run --rm=true"
+        flexy_container_args = self.build_container_args()
+        cmd_string = " ".join([cmd, " ".join(flexy_container_args), 'destroy'])
+        return cmd_string
+
+    def get_log(self):
```

Comment:
Just curious as I remember we had this kind of PR opened for long time : https://github.com/red-hat-storage/ocs-ci/pull/1096/files

This could actually solve the whole issue you would like to achieve here with live log right?

But the way how it's implemented in the PR looks more clean to me.

Reply:
Agree, that PR looks neater. I am not really familiar with ```select``` call, I will try to come up with this.

Reply:
Using ```select``` we had a side effect of ```out of order logs``` hence redirected stderr to stdout itself and reading directly from stdout pipe continuously.